clients/command-kafka/src/main/java/org/eclipse/hono/client/command/kafka/KafkaBasedCommandResponseSender.java

```diff
@@ -76,26 +82,37 @@ public KafkaBasedCommandResponseSender(
             span.log(String.format("using messaging type %s instead of type %s used for the original command",
                     getMessagingType(), response.getMessagingType()));
         }
-        return sendAndWaitForOutcome(
-                topic,
-                response.getTenantId(),
-                response.getDeviceId(),
-                response.getPayload(),
-                getHeaders(response),
-                span)
-            .onComplete(ar -> span.finish());
+
+        return tenantClient.get(response.getTenantId(), context)
+                .compose(tenant -> sendAndWaitForOutcome(
+                    topic,
+                    response.getTenantId(),
+                    response.getDeviceId(),
+                    response.getPayload(),
+                    getHeaders(response, tenant),
+                    span))
+                .onComplete(ar -> span.finish());
     }
 
-    private Map<String, Object> getHeaders(final CommandResponse response) {
-        final Map<String, Object> headers = new HashMap<>(response.getAdditionalProperties());
+    private Map<String, Object> getHeaders(final CommandResponse response, final TenantObject tenant) {
+
+        final var headers = new DownstreamMessageProperties(
+                CommandConstants.NORTHBOUND_COMMAND_RESPONSE_ENDPOINT,
+                tenant.getDefaults().getMap(),
+                null,
```

Comment:
I guess you are right, we should change the `CommandResponseSender` interface to accept a `TenantObject` and a `DeviceRegistration` in its method. That way it will also more resemble the Telemetry and EventSender interfaces.